src/components/compliance/resources/http.rs

```diff
@@ -0,0 +1,173 @@
+use bytes::BytesMut;
+use http::{Method, Request, Uri};
+use hyper::{Body, Client};
+use tokio::sync::mpsc;
+use tokio_util::codec::Encoder;
+use vector_core::event::Event;
+
+use crate::components::compliance::sync::{Configured, ExternalResourceCoordinator, WaitHandle};
+
+use super::{ResourceCodec, ResourceDirection};
+
+/// An HTTP resource.
+pub struct HttpConfig {
```

Comment:
:thinking:  should that be called `HttpResourceDefinition` (or something like that) to not get it mixed up with `<..>Config` used in the sources/sinks etc?